Dockerfile

```diff
@@ -0,0 +1,39 @@
+# Build the source code
+FROM node:16-alpine AS builder
+RUN apk add --no-cache libc6-compat git python3 py3-pip make g++
+WORKDIR /app
+COPY . .
+
+# install deps
+RUN yarn install
+
+# Next.js collects completely anonymous telemetry data about general usage.
+# Learn more here: https://nextjs.org/telemetry
+# Uncomment the following line in case you want to disable telemetry during the build.
+ENV NEXT_TELEMETRY_DISABLED 1
+
+RUN yarn build
+RUN yarn export
+
+
+# Production image, copy out dir and run serve
+FROM node:16-alpine AS runner
+WORKDIR /app
+
+ENV NODE_ENV production
```

Comment:
This should probably go before yarn build?
Also what about all the other env variables? Like the Infura key etc

Reply:
variables are loaded from your `.env` file.

I updated my `NEXT_PUBLIC_INFURA_TOKEN` to make sure that it updates properly and its working fine:

![Captura de pantalla 2022-12-19 a las 10 24 57](https://user-images.githubusercontent.com/26763673/208392869-abb9db85-ad9b-408e-9211-2cfdd5cf8108.png)
![Captura de pantalla 2022-12-19 a las 10 25 35](https://user-images.githubusercontent.com/26763673/208392875-e9f85ba2-8053-4dc7-9603-be42744e9ebd.png)

Reply:
And [the fixed value `ENV NODE_ENV production` in the Dockerfile](https://github.com/vercel/next.js/blob/canary/examples/with-docker/Dockerfile#L37) its seems just to make sure that the node runs in production mode